### PR TITLE
+ fhir examples

### PR DESCRIPTION
**Pull request by ericprud**
related to #131

### CODE REVIEW

File: proposals/primer/index.bs [hunk 1]
```diff
@@ -519,6 +522,63 @@ Issue: Snippet below needs to be updated
...
+  show: 11-100
+  </pre>
+  <figcaption>simplified Care Plan instance data</figcaption>
+</figure>
+
+This CarePlan for <a href="https://bioportal.bioontology.org/ontologies/ICD10/?p=classes&conceptid=http%3A%2F%2Fpurl.bioontology.org%2Fontology%2FICD10%2FE16.2">unspecified hypoglyceamia</a> includes a reference to Bob's home address.
+Anyone overseeing logists for the CarePlan's Appointments will need access to that location.
+This is selected with the ShapePath <code class="ShapePath">@&lt;AppointmentLocation>~fhir:link</code>.
+(If the schema did not provide a specific shape for <code>AppointmentLocation</code>, we would need the long form of the path, i.e. <code class="ShapePath">@&lt;CarePlan>~fhir:activity~fhir:plannedActivityDetail[fhir:kind="Appointment"]~fhir:CodeableReference.reference~fhir:link</code>.)
+The referenced ../Location/HomeAddress contains contact information and addresses:
```
Comment (TallTed) [inline proposals/primer/index.bs:562]:
```suggestion
The referenced <code>../Location/HomeAddress</code> contains contact information and addresses:
```

**Review by TallTed — CHANGES_REQUESTED** [review]
A few random nits

File: proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+PREFIX interop: <http://www.w3.org/ns/solid/interop#>
+PREFIX solidtrees: <https://solidshapes.example/trees/>
+BASE <https://bob.example/>
+
+</system/registrations/A-fhir-CarePlan-DataRegistration-F>
+  a interop:DataRegistration ;
+  interop:registeredBy <https://bob.example/#id> ;
+  interop:registeredWith <https://solidmin.example/#app> ;
+  interop:registeredAt "2019-03-14T13:42:32.000Z"^^xsd:dateTime ;
+  interop:registeredShapeTree fhirshapes:CarePlanContainer ;
```
Comment (ericprud) [inline proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl:11]: i pointed to the data reg at CarePlanContainer. Is that correct?

File: proposals/primer/index.bs [hunk 3]
```diff
@@ -571,20 +581,58 @@ This is selected with the ShapePath <code class="ShapePath">@&lt;AppointmentLoca
...
+  show: 5-100
+  </pre>
+  <figcaption>simplified Care Plan ShapeTree Locator</figcaption>
+</figure>
+
+#### Following ShapePath references for access controls
+
+An authentication agent can now allow a user to set permissions on <code>A-CarePlan-F/A-HypoglycemiaManagement-F</code> and extend those permissions to the referenced <code>A-Location-F/A-HomeAddress-F</code>:
+
+1. A HEAD on <code>A-CarePlan-F/A-HypoglycemiaManagement-F</code> includes a Link: header rel=<code>http://shapetrees.org/#ShapeTreeLocator</code> pointing to <code>A-HypoglycemiaManagement-F?stloc#locator</code>.
```
Comment (TallTed) [inline proposals/primer/index.bs]:
```suggestion
1. A <code>HEAD</code> on <code>A-CarePlan-F/A-HypoglycemiaManagement-F</code> includes a <code>Link:</code> header <code>rel</code>=<code>http://shapetrees.org/#ShapeTreeLocator</code> pointing to <code>A-HypoglycemiaManagement-F?stloc#locator</code>.
```

File: proposals/primer/index.bs [hunk 4]
```diff
@@ -519,6 +522,143 @@ Issue: Snippet below needs to be updated
...
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>simplified Location for Bob</figcaption>
+</figure>
+
+Anyone overseeing logistics for the CarePlan's Appointments will need access to that location.
+This is selected with the ShapePath <code class="ShapePath">@&lt;AppointmentLocation> ~ fhir:link</code>.
+(If the schema did not provide a specific shape for <code>AppointmentLocation</code>,
+we would need to start a path from a defined shape, e.g.
```
Comment (TallTed) [inline proposals/primer/index.bs:581]:
```suggestion
we would need to start a path from a defined shape, e.g.,
```

File: proposals/primer/index.bs [hunk 5]
```diff
@@ -519,6 +522,143 @@ Issue: Snippet below needs to be updated
...
+    <pre class=include-code>
+      path: snippets/bob.example/fhir/CarePlan/HypoglycemiaManagement.ttl
+      highlight: shex
+      show: 43-45
+    </pre>
+     matches the above <code>fhir:link</code> triple constraint.
+
+#### Clinical data registration
+
+The CarePlan data registration would reference the <code>/data/A-CarePlan-F/<code> container which includes <code>A-HypoglycemiaManagement-F</code> and <code>A-Wellness-F</code> mentioned above:
```
Comment (elf-pavlik) [inline proposals/primer/index.bs:651]:
```suggestion
The CarePlan data registration would reference the <code>/data/A-CarePlan-F/</code> container which includes <code>A-HypoglycemiaManagement-F</code> and <code>A-Wellness-F</code> mentioned above:
```

File: proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+PREFIX interop: <http://www.w3.org/ns/solid/interop#>
+PREFIX solidtrees: <https://solidshapes.example/trees/>
+BASE <https://bob.example/>
+
+</system/registrations/A-fhir-CarePlan-DataRegistration-F>
+  a interop:DataRegistration ;
+  interop:registeredBy <https://bob.example/#id> ;
+  interop:registeredWith <https://solidmin.example/#app> ;
+  interop:registeredAt "2019-03-14T13:42:32.000Z"^^xsd:dateTime ;
+  interop:registeredShapeTree fhirshapes:CarePlan ;
```
Comment (elf-pavlik) [inline proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl:11]:
```suggestion
  interop:registeredShapeTree fhirshapeTree:CarePlan ;
```

**Comment by elf-pavlik** [comment]
Looks good @ericprud, thank you! Could you please also add Data Registration for `fhirshapeTree:Location`

**Comment by ericprud** [comment]
> Looks good @ericprud, thank you! Could you please also add Data Registration for `fhirshapeTree:Location`?

I think that would imply that we need a Shape Tree Locator, a Shape Tree, and finally a schema for it. I wonder where to draw the line.

> I also don't understand why do you need `fhirshapeTree:CarePlanContainer` if it doesn't have any Data Registration and Data Instance?

It provides the ShapeTree for <http://bob.example/data/A-CarePlan-F/> and is referenced as the root of the locator.

File: proposals/primer/index.bs (hunk 3 above)
Comment (ericprud) [reply proposals/primer/index.bs]: this is now in a <pre/>

File: proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+PREFIX solidtrees: <https://solidshapes.example/trees/>
+BASE <https://bob.example/>
+
+</system/registrations/A-fhir-CarePlan-DataRegistration-F>
+  a interop:DataRegistration ;
+  interop:registeredBy <https://bob.example/#id> ;
+  interop:registeredWith <https://solidmin.example/#app> ;
+  interop:registeredAt "2019-03-14T13:42:32.000Z"^^xsd:dateTime ;
+  interop:registeredShapeTree fhirshapeTree:CarePlan ;
+  interop:hasRegisteredDataInstance
```
Comment (elf-pavlik) [inline proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl:12]: We should clarify if we use this on `ldp:contains`, discussing it now during the call with @justinwb

Comment (elf-pavlik) [reply proposals/primer/snippets/bob.example/fhir-CarePlan-DataRegistration.ttl:12]:
```suggestion
  ldp:contains
```

I think we should use this one at least for now, I may create separate issue to discuss it futher

**Comment by elf-pavlik** [comment]
@ericprud @justinwb I think we should merge this one later today. We can always create issue to add missing data registration. BTW we could at least add snippet for data registration without including it in primer document.

**Comment by ericprud** [comment]
I added the Location ShapeTreeLocator and DataRegistration. Should be good to go, though the preview isn't showing up ('cause I'm so impatient)

**Pull request merged**